### ISSUE TITLE
[Request] Add native Linux support

### ISSUE THREAD

**Issue by RagdollClash** [thesupersonic16/HedgeModManager#568]
## This is being made it's own issue due to how major it is.

As of recently, HedgeModManager has been slowly becoming nonfunctional on Linux (especially the Steam Deck) to the point where we're now getting reports that the program as a whole no longer launches.

The manager no longer launching can be fixed by adding native Linux support to HedgeModManager, although code support is another story.

**Comment by Mash0Star** [comment]
I agree with this. I can't open up frontiers anymore since the mod loader breaks the game from launching on Linux and the Steam Deck. A native Linux version would be great. I'm sorry i can't do anything to make it myself since i don't have the knowledge or expertise to do such a thing.

**Comment by Joefers** [comment]
Hello! Is this issue being worked? 

I've spent an inordinate amount of time getting HMM running on my Steam Deck for Frontiers (finally successfully). I think I can take a stab at making an .AppImage build, but the key things from what I can tell are:
- Installing dotnet40 and dotnet48 into the prefix for Frontiers, and somehow checking for this. Without these installed, Frontiers simply refuses to boot, presumably since it can't properly load the `d3d11.dll` file.
- Handling building the C# .sln, then packaging that into an .AppImage.

How are your builds for releases handled? I found a script written that appears to handle this specific use case, so I was going to see if I could get results from it: https://github.com/kuiperzone/Publish-AppImage

I'll update if I get any progress. Cheers.

**Comment by Joefers** [comment]
Some updates...

It's looking like WPF (**_Windows_** Presentation Foundation) applications can't run on Linux (without Wine or some other Windows technology).

That leaves us with these options:
- Force the issue by implementing a cross-platform WPF replacement such as [Avalonia](https://avaloniaui.net/). This way the thing will work out of the box on Linux. This is probably The Better but Takes More Time Approach. Alternatively we could investigate if Avalonia could be used only for the Linux builds.
- Stay with WPF, meaning no native Linux version, and staying with the Windows .exe and Wine to run it. This is what's being done already -- HMM appears to create a desktop shortcut for running a configured Bottle, which can then run HMM.

@thesupersonic16 I see that there was a config for Bottles [here](https://github.com/thesupersonic16/HedgeModManager/commit/f30bbf1995307d4918ddc56b7b6f88d27a7abbad). Did you have success with it? I see in SteamTinkerLaunch they generate a custom prefix for HMM with the dotnet packages installed -- wouldn't the Bottle already accomplish this?

I'm working on a new Github Action in my fork for generating a Linux build that can be dropped to a Steam Deck and "Just Runs". I haven't been able to get a good build yet but I'll keep at it.

**Comment by thesupersonic16** [comment]
The bottles config worked on my system the last time I booted it. Only issue I had was getting dotnet48 on each d3d11 game which needed winetricks. I did hear some reports on it requiring an older proton version when using protontricks.

Also STL doesn't work on my system, haven't gotten around to looking into why.

**Comment by Joefers** [comment]
Roger that, thank you. That lines up with my experiences too -- I had to use the protontricks GUI to install dotnet40 and dotnet48 successfully into the HMM and Sonic Frontiers prefix. For whatever reason the STL calls to do so would fail out.

I'll start looking into an Avalonia implementation and update if I get any traction.

**Comment by igeljaeger** [comment]
getting this to work on linux is possible but a huge hassle. Esp. on the Steam Deck. I will just play my games unmodded until this gets either an easy to use proton focused version or a native linux version. We exist.

**Comment by lolcatplays** [comment]
With the recent update the application works well as installed by steamtinkerlaunch. The new mod loader is borked for HE2 games however.

**Comment by thesupersonic16** [comment]
We might need to fix the registry patcher to use dinput8.

**Comment by declanmcmains** [comment]
i think a native Linux port would be great and give more accessibility and it would allow me to mod sonic generation's

**Comment by shrobbyy** [comment]
I think the current ways to use HMM on Linux are way too complicated. I also tried the guide recommended here but it didn't work. In fact, it made my game unable to launch at all, giving me a Denuvo error. It is possible to add the HMM executable onto Steam via adding it as a 3rd party application. It will detect all games, the mod manager works all fine, but applying mods and codes does not anything and the game stays vanilla.

The SADX and SA2 mod loaders do work perfectly fine that way. I also were able to use Sonic Unleashed on Xenia with the Unleashed Mod Manager, which both don't have Linux builds and have to be launched via Wine or Proton, in the same way I do for Sonic Adventure.
So it would be great if HMM would **at least** work that way.
